### PR TITLE
update rpc tests

### LINKED ISSUE #263: Improve tests for GetLogs, GetFilterChanges
Currently the tests for the rpc are not very reliable, this is more evident now with #257 and once #259 is merged we would like to add the it-tests to travis build and it should block pr if not passing.

More details about the errors:
```
=== RUN   TestEth_GetLogs_Topics_AB
--- FAIL: TestEth_GetLogs_Topics_AB (4.02s)
    rpc_test.go:520: 
                Error Trace:    rpc_test.go:520
                Error:          Not equal: 
                                expected: 1
                                actual  : 0
                Test:           TestEth_GetLogs_Topics_AB


--- FAIL: TestEth_GetFilterChanges_Topics_XB (7.02s)
    rpc_test.go:481: 
                Error Trace:    rpc_test.go:481
                Error:          Not equal: 
                                expected: 1
                                actual  : 0
                Test:           TestEth_GetFilterChanges_Topics_XB

--- FAIL: TestEth_GetFilterChanges_Topics_AB (8.02s)
    rpc_test.go:444: 
                Error Trace:    rpc_test.go:444
                Error:          Not equal: 
                                expected: 1
                                actual  : 0
                Test:           TestEth_GetFilterChanges_Topics_AB
```

### PR DESCRIPTION
**Pull request by noot**
Closes: #263 

## Description

- add waitForReceipt function to wait until transaction is successfully included, then look for logs
- rename GetTxLogs to GetTransactionLogs

______

For contributor use:

- [x] Targeted PR against correct branch (see [CONTRIBUTING.md](https://github.com/cosmos/cosmos-sdk/blob/master/CONTRIBUTING.md#pr-targeting))
- [x] Linked to Github issue with discussion and accepted design OR link to spec that describes this work.
- [x] Code follows the [module structure standards](https://github.com/cosmos/cosmos-sdk/blob/master/docs/building-modules/structure.md).
- [x] Wrote unit and integration [tests](https://github.com/cosmos/cosmos-sdk/blob/master/CONTRIBUTING.md#testing)
- [ ] Updated relevant documentation (`docs/`) or specification (`x/<module>/spec/`)
- [ ] Added  relevant `godoc` [comments](https://blog.golang.org/godoc-documenting-go-code).
- [ ] Added a relevant changelog entry to the `Unreleased` section in `CHANGELOG.md`
- [ ] Re-reviewed `Files changed` in the Github PR explorer

### CODE REVIEW

File: tests/rpc_test.go [hunk 1]
```diff
@@ -94,26 +94,22 @@ func call(t *testing.T, method string, params interface{}) (*Response, error) {
 	/* #nosec */
 	res, err := http.Post(ETHERMINT_NODE_HOST, "application/json", bytes.NewBuffer(req))
 	if err != nil {
```
Comment (fedekunze) [inline tests/rpc_test.go]: these should all be `require.NoError(t, err)` instead of the error checks

**Review by fedekunze — CHANGES_REQUESTED** [review]
LGTM, 1 minor change required

File: tests/rpc_test.go (hunk 1 above)
Comment (noot) [reply tests/rpc_test.go]: done

**Review by fedekunze — APPROVED** [review]
ACK

**Pull request merged**